pkg/kyverno/common/common.go

```diff
@@ -372,6 +372,9 @@ func GetVariable(variablesString, valuesFile string, fs billy.Filesystem, isGit
 		for _, p := range values.Policies {
 			resourceMap := make(map[string]Resource)
 			for _, r := range p.Resources {
+				if r.Values["request.operation"] == "" {
```

Comment:
You are basically there, you need to check if the `key` `request.operation` exists in `r.Values` OR if it's an empty string.

So (written from memory - just a starting point):
```
if val, ok := r.Values["request.operation"]; ok || val == "" {
    //do something here
}
```

Reply:
> 
> 
> You are basically there, you need to check if the `key` `request.operation` exists in `r.Values` OR if it's an empty string.
> 
> So (written from memory - just a starting point):
> 
> ```
> if val, ok := r.Values["request.operation"]; ok || val == "" {
>     //do something here
> }
> ```

suppose `request.operation` doesn't exit in the `values.yaml` file and at the same time  `request.operation` is present in the policy, then in that case it will become compulsory to check whether the policy contains `request.operation` . And if it do so, then it need to add key and value i.e.  `request.operation` equals `CREATE`  manually or explicity in the `values.yaml` file.

Reply:
> 
> 
> You are basically there, you need to check if the `key` `request.operation` exists in `r.Values` OR if it's an empty string.
> 
> So (written from memory - just a starting point):
> 
> ```
> if val, ok := r.Values["request.operation"]; ok || val == "" {
>     //do something here
> }
> ```

And the above code would be something like:- 
```
for _, r := range p.Resources {
if v, ok := r.Values["request.operation"]; ok {
	if v == "" {
		r.Values["request.operation"] = "CREATE"
	}
	
}

Reply:
I don't know but it happens many times with me. As i start discussing thing with others the solutions start comming to mind. And finally I got solution for the above also. 
Now it will work for the case if user only provides,
```
policies:
- name: restrict-pod-counts
  rules:
    - name: restrict-pod-count
      values:
        podcounts: "40"
  resources:
  - name: myapp-pod
```

```
Executing restrict-pod-count...
valuesMap:  map[restrict-pod-counts:map[myapp-pod:{myapp-pod map[request.operation:CREATE]}]]

applying 1 policy to 1 resource...

│───│─────────────────────│────────────────────│───────────────────────│────────│
│ # │ POLICY              │ RULE               │ RESOURCE              │ RESULT │
│───│─────────────────────│────────────────────│───────────────────────│────────│
│ 1 │ restrict-pod-counts │ restrict-pod-count │ default/Pod/myapp-pod │ Pass   │
│───│─────────────────────│────────────────────│───────────────────────│────────│
```